tooling/ssa_fuzzer/fuzzer/fuzz_targets/field.rs

```diff
@@ -0,0 +1,454 @@
+) -> bool {
+    index_presented(first_index, acir_witnesses_indeces, brillig_witnesses_indeces)
+        && index_presented(second_index, acir_witnesses_indeces, brillig_witnesses_indeces)
+}
+
+/// Represents an array in the SSA
+struct Array {
+    id: Id<Value>,
+    length: u32,
+}
+
+/// Main context for the fuzzer containing both ACIR and Brillig builders and their state
+/// It works with indices of variables Ids, because it cannot handle Ids logic for ACIR and Brillig
+struct FuzzerContext {
+    /// ACIR builder
+    acir_builder: FuzzerBuilder,
+    /// Brillig builder
+    brillig_builder: FuzzerBuilder,
+    /// Ids of ACIR witnesses stored as u32
+    acir_ids: Vec<u32>,
+    /// Ids of Brillig witnesses stored as u32
+    brillig_ids: Vec<u32>,
+    /// ACIR arrays
+    acir_arrays: Vec<Array>,
+    /// Brillig arrays
+    brillig_arrays: Vec<Array>,
+}
+
+impl FuzzerContext {
+    /// Creates a new fuzzer context with the given type
+    fn new(type_: Type) -> Self {
+        let mut acir_builder = FuzzerBuilder::new_acir();
+        let mut brillig_builder = FuzzerBuilder::new_brillig();
+        acir_builder.insert_variables(type_.clone());
+        brillig_builder.insert_variables(type_.clone());
+        let mut acir_ids = vec![];
+        let mut brillig_ids = vec![];
+        // by default private variables ids are indexed from 0 to NUMBER_OF_VARIABLES_INITIAL
+        for i in 0..config::NUMBER_OF_VARIABLES_INITIAL {
+            acir_ids.push(i);
+            brillig_ids.push(i);
+        }
+        Self {
+            acir_builder,
+            brillig_builder,
+            acir_ids,
+            brillig_ids,
+            acir_arrays: vec![],
+            brillig_arrays: vec![],
+        }
+    }
+
+    /// Creates a new array from a vector of indices of variables
+    /// Skips non-presented variables
+    fn insert_array(&mut self, elements: Vec<u32>) {
+        let mut acir_values_ids = vec![];
+        let mut brillig_values_ids = vec![];
+        for elem in elements {
+            if !index_presented(
+                elem,
+                &mut self.acir_ids,
+                &mut self.brillig_ids,
+            ) {
+                continue;
+            }
+            acir_values_ids.push(elem);
+            brillig_values_ids.push(elem);
+        }
+        let acir_len = acir_values_ids.len();
+        let brillig_len = brillig_values_ids.len();
+        let acir_array = self.acir_builder.insert_make_array(acir_values_ids);
+        let brillig_array = self.brillig_builder.insert_make_array(brillig_values_ids);
+        self.acir_arrays.push(Array { id: acir_array, length: acir_len as u32 });
+        self.brillig_arrays.push(Array { id: brillig_array, length: brillig_len as u32 });
+    }
+
+    /// Gets an element from an array at the given index
+    fn insert_array_get(&mut self, array_idx: u32, index: u32) {
+        if array_idx >= self.acir_arrays.len() as u32
```

Comment:
It is better to just use indices modulo lengths. Otherwise the majority of testcases will skip instructions instead of creating them